trunner/dut.py

```diff
@@ -202,6 +202,43 @@ def open(self):
         # on proper guest OSes this results in `\r\r\n` which might be incorrectly interpreted by CI log viewers
         # use custom preexec_fn to setup termios of child PTY
         self.pexpect_proc = pexpect.spawn(*self.args, preexec_fn=self._set_termios_raw, **self.kwargs)
+        time.sleep(0.2)
+        self._set_logfiles()
+
+
+class fdspawncustom(pexpect.spawn):
+    """
+    The current UART implementation on the ARMv8R52MPS3AN536 breaks when bytes are sent too fast.
+    To alleviate this problem, we override the pexpect class and introduce a delay in the send function.
+    """
+
+    def send(self, s):
+        ret = 0
+        for c in s:
+            ret += super().send(c)
+            time.sleep(0.03)
+
+        return ret
+
+
+class ARMv8R52MPS3AN536_Dut(ProcessDut):
+    @staticmethod
+    def _set_termios_raw():
+        """set current tty not to process newlines"""
+        attr = termios.tcgetattr(STDOUT_FILENO)
+
+        # disable any newline manilpulations in c_oflag
+        attr[1] &= ~(termios.ONLCR | termios.OCRNL | termios.ONOCR | termios.ONLRET)
+
+        termios.tcsetattr(STDOUT_FILENO, termios.TCSANOW, attr)
+
+    def open(self):
+        # qemu when using stdio serial enforces ONLCR termios flag (converting `\n` to `\r\n`)
+        # on proper guest OSes this results in `\r\r\n` which might be incorrectly interpreted by CI log viewers
+        # use custom preexec_fn to setup termios of child PTY
+        self.pexpect_proc = fdspawncustom(
+            *self.args, preexec_fn=self._set_termios_raw, **self.kwargs
+        )
         self._set_logfiles()
```

Comment:
Why don't inherit from QemuDut?
```suggestion
class ARMv8R52MPS3AN536_Dut(QemuDut):
    def open(self):
        # qemu when using stdio serial enforces ONLCR termios flag (converting `\n` to `\r\n`)
        # on proper guest OSes this results in `\r\r\n` which might be incorrectly interpreted by CI log viewers
        # use custom preexec_fn to setup termios of child PTY
        self.pexpect_proc = fdspawncustom(
            *self.args, preexec_fn=self._set_termios_raw, **self.kwargs
        )
        self._set_logfiles()
```